src/cache_invalidate.c

```diff
@@ -108,30 +167,52 @@ PGDLLEXPORT Datum invalidate_relcache(PG_FUNCTION_ARGS);
 PG_FUNCTION_INFO_V1(invalidate_relcache);
 
 /*
- *	This is similar to invalidate_relcache_trigger but not a trigger.
- *	Not used regularly but useful for debugging.
+ * Force a cache invalidation for a catalog table.
+ *
+ * This function is used for debugging purposes and triggers
+ * a cache invalidation.
  *
+ * The first argument should be the catalog table that has changed, warranting a
+ * cache invalidation.
  */
 Datum
 invalidate_relcache(PG_FUNCTION_ARGS)
 {
-	Oid			proxy_oid = PG_GETARG_OID(0);
-
-	/* arg 0 = relid of the cache_inval_proxy table */
-	CacheInvalidateRelcacheByRelid(proxy_oid);
+	cache_invalidate_catalog(PG_GETARG_OID(0), CMD_UPDATE);
+	PG_RETURN_BOOL(true);
+}
 
-	/* tuple to return to executor */
-	return BoolGetDatum(true);
+static void
+cache_invalidate_xact_end(XactEvent event, void *arg)
+{
+	switch (event)
+	{
+		case XACT_EVENT_ABORT:
+		case XACT_EVENT_PARALLEL_ABORT:
+
+			/*
+			 * Invalidate caches on aborted transactions to purge entries that
+			 * have been added during the transaction and are now no longer
+			 * valid. Note that we need not signal other backends of this
+			 * change since the transaction hasn't been committed and other
+			 * backends cannot have the invalid state.
+			 */
+			hypertable_cache_invalidate_callback();
```

Comment:
A locally aborted transaction has no effect on other backend's caches since changes haven't been committed, so this  is a local invalidation event that only concerns the current backend's cache.

Reply:
In any case, not sure calling `cache_invalidate_callback()` with the relid of the hypertable catalog is less weird since I still have to list the hypertable cache here by catalog relid. I could do that if you feel that roundabout way via relid is better for some reason

Reply:
I think calling cache_invalidate_callback() with an argument of 0 (Invalid OID) would cause all caches to be invalidated.

Reply:
I cannot see this logic in the code for `cache_invalidate_callback()`. I can, of course, implement that, but not sure what the benefit is. It just makes things more obscure to have to call that function with an OID Datum set to `InvalidOid` to invalidate all caches.

If this is a about clarity, I can create a wrapper function called `cache_invalidate_all()` or something that would invalidate all caches. But now we have only one cache to invalidate so it seems somewhat pointless to do that.